rotkehlchen/exchanges/binance.py

```diff
@@ -891,8 +900,170 @@ def query_online_trade_history(
 
             trades.append(trade)
 
+        fiat_payments = self._query_online_fiat_payments(start_ts=start_ts, end_ts=end_ts)
+        if fiat_payments:
+            trades += fiat_payments
+            trades.sort(key=lambda x: x.timestamp)
+
         return trades
 
+    def _query_online_fiat_payments(self, start_ts: Timestamp, end_ts: Timestamp) -> List[Trade]:
+        fiat_buys = self._api_query_list_within_time_delta(
+            start_ts=start_ts,
+            end_ts=end_ts,
+            time_delta=API_TIME_INTERVAL_CONSTRAINT_TS,
+            api_type='sapi',
+            method='fiat/orders',
+            additional_options={'transactionType': 0},
+        )
+        log.debug(f'{self.name} fiat buys history result', results_num=len(fiat_buys))
+        fiat_sells = self._api_query_list_within_time_delta(
+            start_ts=start_ts,
+            end_ts=end_ts,
+            time_delta=API_TIME_INTERVAL_CONSTRAINT_TS,
+            api_type='sapi',
+            method='fiat/orders',
+            additional_options={'transactionType': 1},
+        )
+        log.debug(f'{self.name} fiat sells history result', results_num=len(fiat_sells))
+
+        trades = []
+        for idx, raw_fiat in enumerate(fiat_buys + fiat_sells):
+            trade_type = TradeType.BUY if idx < len(fiat_buys) else TradeType.SELL
+            trade = self._deserialize_fiat_payment(raw_fiat, trade_type=trade_type)
+            if trade:
+                trades.append(trade)
+
+        return trades
+
+    def _deserialize_fiat_payment(
+            self,
+            raw_data: Dict[str, Any],
+            trade_type: TradeType,
+    ) -> Optional[Trade]:
+        """Processes a single deposit/withdrawal from binance and deserializes it
+
+        Can log error/warning and return None if something went wrong at deserialization
+        """
+        try:
+            if 'status' not in raw_data or raw_data['status'] != 'Successful':
```

Comment:
There were errors in deserialization here. Remember. When a buy is made, then the BASE asset is bought with QUOTE asset. When a sell is made then the BASE asset is sold for QUOTE asset. This should now be okay.

Also fiat payments have a price field which is what should be used.